src/Microsoft.FileFormats/Pointer.cs

```diff
@@ -88,16 +94,19 @@ public class SizeTPointerLayout : PointerLayout
 
         public override object Read(IAddressSpace dataSource, ulong position)
         {
-            ulong val = (SizeT)_storageLayout.Read(dataSource, position);
+            ulong value = (SizeT)StorageLayout.Read(dataSource, position);
             Pointer p = (Pointer)Activator.CreateInstance(Type);
-            p.Init(_targetLayout, val);
+            p.Initialize(value, this);
```

Comment:
Could we leave the Pointer class alone and invoke this as?
```C#
p.Init(TargetLayout, value);
```

I think in order to prevent recursion we have to avoid resolving the target layout until after the PointerLayout is constructed and added to the LayoutManager, but at the point we are reading an instance of the pointer from memory we've deferred long enough right?

Reply:
Yes, you are right of course. Delaying it until the Pointer is created is enough.  I backed out the changes to Pointer/Pointer<,,> and did as you suggested. Thanks.